runtime/browser/xwalk_content_base.cc

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) 2015 Intel Corporation. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "xwalk/runtime/browser/xwalk_content_base.h"
+
+#include <string>
+
+#include "base/prefs/pref_service.h"
+#include "components/autofill/content/browser/content_autofill_driver_factory.h"
+#include "components/autofill/core/browser/autofill_manager.h"
+#include "components/user_prefs/user_prefs.h"
+#include "content/public/browser/render_view_host.h"
+#include "content/public/common/renderer_preferences.h"
+#include "xwalk/runtime/browser/xwalk_autofill_client.h"
+#include "xwalk/runtime/browser/xwalk_content_browser_client.h"
+#include "xwalk/runtime/browser/xwalk_browser_context.h"
+#include "xwalk/runtime/browser/xwalk_runner.h"
+
+#if defined(OS_ANDROID)
+#include "base/android/locale_utils.h"
+#include "xwalk/runtime/browser/android/xwalk_autofill_client_android.h"
+#include "xwalk/runtime/browser/android/xwalk_content.h"
+#endif
+
+#if !defined(OS_ANDROID)
+#include "xwalk/runtime/browser/xwalk_autofill_client_desktop.h"
+#endif
+
+namespace xwalk {
+
+const void* kXWalkContentUserDataKey = &kXWalkContentUserDataKey;
+
+class XWalkContentUserData : public base::SupportsUserData::Data {
+ public:
+  explicit XWalkContentUserData(XWalkContentBase* ptr) : content_(ptr) {}
+  static XWalkContentBase* GetContents(content::WebContents* web_contents) {
+    if (!web_contents)
+      return nullptr;
+    XWalkContentUserData* data = reinterpret_cast<XWalkContentUserData*>(
+      web_contents->GetUserData(kXWalkContentUserDataKey));
+    return data ? data->content_ : nullptr;
+  }
+
+ private:
+  XWalkContentBase* content_;
+};
+
+XWalkContentBase* XWalkContentBase::GetContents(
+    content::WebContents* web_contents) {
+  return XWalkContentUserData::GetContents(web_contents);
+}
+
+XWalkContentBase::~XWalkContentBase() {
+}
+
+XWalkContentBase::XWalkContentBase(
+    scoped_ptr<content::WebContents> web_contents)
+    : web_contents_(web_contents.Pass()) {
+#if defined(OS_ANDROID)
```

Comment:
What about we rename XWalkContentBase to XWalkContentManager (you have better idea?) and make it a member in both XWalkContent (Android) and Runtime (Desktop).